### PR TITLE
[FEATURE] Document how the ViewHelper reference is generated

### PR DESCRIPTION
**Pull request by linawolf**

### CODE REVIEW

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+=====================================
+
+The PHP files defining the ViewHelpers are transferred into :file:`schema.xsd`
+files. The package `fluid-schema-generator <https://github.com/TYPO3/Fluid.SchemaGenerator>`__
+is responsible for this step. So if information is already missing in the
+:file:`schema.xsd`, adjustments need to be made here. The information from the
+:file:`schema.xsd` files is also used by IDEs for context sensitive help in
+editors etc.
+
+You can regenerate the :file:`schema.xsd` files for local testing like this:
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:30]:
```suggestion
You can generate the :file:`schema.xsd` files yourself for local testing, like this:
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+..  code-block:: bash
+
+    git clone git@github.com:typo3/typo3.git core
+    cd core
+    composer require -o -n --no-progress typo3/fluid-schema-generator "^2.1"
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:36]: Should we use a version here, that we may forget to update? Maybe just use "*" to use the most recent one...?

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    mkdir -p ../schemas/typo3fluid/fluid/latest
+    ./bin/generateschema TYPO3Fluid\\\Fluid > ../schemas/typo3fluid/fluid/latest/schema.xsd
+    mkdir -p ../schemas/typo3/core/latest
+    ./bin/generateschema TYPO3\\\CMS\\\Core > ../schemas/typo3/core/latest/schema.xsd
+    mkdir -p ../schemas/typo3/fluid/latest
+    ./bin/generateschema TYPO3\\\CMS\\\Fluid > ../schemas/typo3/fluid/latest/schema.xsd
+    mkdir -p ../schemas/typo3/backend/latest
+    ./bin/generateschema TYPO3\\\CMS\\\Backend > ../schemas/typo3/backend/latest/schema.xsd
+
+The TYPO3 Core will automatically install Fluid Rendering Engine in the version
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:46]:
```suggestion
The composer step will automatically require the Fluid Rendering Engine in the version
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+used by that Core version.
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:47]:
```suggestion
used by the Core version you fetched via GIT (by default, `main`).
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+..  _fluid-viewhelper-reference-generation-rst:
+
+Generation of the reSTructured text files
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:51]:
```suggestion
Generation of the reStructuredText files
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+=========================================
+
+The :abbr:`rst (reSTructured text)` files get generated from the
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:54]:
```suggestion
The :abbr:`rst (reStructuredText)` files get generated from the
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+:file:`schema.xsd` files (See section
+:ref:`fluid-viewhelper-reference-generation-schema`) with the help of the
+following tool:
+`Fluid ViewHelper Documentation Generator <https://github.com/TYPO3-Documentation/fluid-documentation-generator>`.
+
+So if the desired information is contained in the :file:`schema.xsd` files but
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:60]:
```suggestion
If any of the ViewHelper source code documentation is only contained in the :file:`schema.xsd` files but
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+not the rst sources, the problem might have to be fixed here.
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:61]:
```suggestion
not the generated standalone :file:`.rst` files, a bug in that generator may exist and needs adressing.
```

Do you have an idea of an example what kind of errors could exist here, that would create wrong RST from schema.xsd?

Also maybe one more sentence about the thing we talked today, something like this maybe:


```
Please note that the TYPO3 Core initially uses `reStructuredText` formatting 
inside the phpDoc comments already,  and this is passed on to the :file:`schema.xsd` 
files with exactly that formatting. 
IDEs may want to interpret  these schemas with expected HTML or MarkDown syntax, 
so you may see raw `reStructuredText` output in this case. 
The generated Documentation for `docs.typo3.org` takes care of 
transforming the generated  :file:`.rst` files to HTML, by utilizing the
`https://github.com/TYPO3-Documentation/render-guides`__ project.
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+..  _fluid-viewhelper-reference-generation-html:
+
+Rendering the ViewHelper reference to HTML
+==========================================
+
+The :abbr:`rst (reSTructured text)` files generated in step
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:68]:
```suggestion
The :abbr:`rst (reStructuredText)` files generated in step
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+:ref:`fluid-viewhelper-reference-generation-rst` will be saved by the
+:ref:`GitHub action <fluid-viewhelper-reference-generation-github-action>` into
+repository `Fluid ViewHelper Reference <https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-ViewHelper>`.
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:71]:
```suggestion
the repository `Fluid ViewHelper Reference <https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-ViewHelper>`.
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+:ref:`guides.xml <settings-guides-xml>`. This repository is then rendered by
+the standard rendering process.
+
+..  _fluid-viewhelper-reference-generation-github-action:
+
+GitHub action "Fluid ViewHelper Documentation"
+==============================================
+
+All processes described above are combined for automatic execution as a GitHub
+action in the repository `t3docs-ci-deploy <https://github.com/TYPO3-Documentation/t3docs-ci-deploy>`__.
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:85]:
```suggestion
action in the repository `t3docs-ci-deploy <https://github.com/TYPO3-Documentation/t3docs-ci-deploy>`__.
It is triggered automatically once a day, or can be executed manually through the GitHub UI by the TYPO3 Documentation team.
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+Watch for failed or stuck `Actions <https://github.com/TYPO3-Documentation/t3docs-ci-deploy/actions>`__.
```
Comment (garvinhicking) [inline Documentation/Maintainers/FluidViewHelper.rst:87]:
```suggestion
Maintainers needs to occasionally watch for failed or stuck `Workflow runs <https://github.com/TYPO3-Documentation/t3docs-ci-deploy/actions>`__.
```

**Review by garvinhicking — COMMENTED** [review]
Thanks so much, I think it is very vital that we have this. Once it's merged, we should link to this section from the mentioned projects (Fluid.Schemagenerator, fluid-documentation-generator, maybe also ci-deploy?) from their READMEs, so that future us remembers where to find this :)

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+..  _fluid-viewhelper-reference-generation-schema:
+
+The Fluid ViewHelper schema generator
+=====================================
+
+The PHP files defining the ViewHelpers are transferred into :file:`schema.xsd`
+files. The package `fluid-schema-generator <https://github.com/TYPO3/Fluid.SchemaGenerator>`__
+is responsible for this step. So if information is already missing in the
+:file:`schema.xsd`, adjustments need to be made here. The information from the
+:file:`schema.xsd` files is also used by IDEs for context sensitive help in
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Maintainers/FluidViewHelper.rst]: I don't think this is our concern. All we have to know is that it is used in other contexts, too

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+The :ref:`Fluid ViewHelper Reference <t3viewhelper:start>` gets automatically
+updated by changes to the according ViewHelper classes in the
+`TYPO3 Core <https://github.com/TYPO3/typo3>`__ and
+the package `Fluid Rendering Engine <https://github.com/TYPO3/Fluid>`__.
+
+The generated documentation specifically depends on the phpDoc-style inline
+comments on top of the ViewHelper classes as well as the configured arguments in their
+php:`initializeArguments()` method.
+
+Context changes need to be made in those two repositories. For contributions
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Maintainers/FluidViewHelper.rst:17]: No that was not by intend, I was sure I commited them

Comment (linawolf) [inline Documentation/Maintainers/FluidViewHelper.rst:17]:
```suggestion
Changes in wording or arguments thus need to be made inside the relevant files of these two repositories. For contributions
```

File: Documentation/Maintainers/FluidViewHelper.rst [hunk 15]
```diff
@@ -0,0 +1,87 @@
+.. include:: /Includes.rst.txt
+..  _fluid-viewhelper-reference-generation:
+
+=====================================
+Fluid ViewHelper reference generation
+=====================================
+
+The :ref:`Fluid ViewHelper Reference <t3viewhelper:start>` gets automatically
+updated by changes to the according ViewHelper classes in the
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Maintainers/FluidViewHelper.rst:9]: Moved the information in an extra sentence

File: Documentation/Maintainers/FluidViewHelper.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/Maintainers/FluidViewHelper.rst]: committed them now

**Comment by garvinhicking** [comment]
Thanks!! :-)

**Pull request merged**